addon/adapters/moderator.js

```diff
@@ -0,0 +1,26 @@
+import OsfAdapter from './osf-adapter';
+
+export default OsfAdapter.extend({
+    pathForType() {
+        return 'preprint_providers';
+    },
+    buildURL(modelName, id, snapshot, requestType, query) {
+        let provider = '';
+        if (query) {
+            provider = query.provider;
+            delete query.provider;
+        } else if (requestType === 'createRecord' && snapshot.record.provider) {
+            provider = snapshot.record.provider;
+            delete snapshot.record.provider;
+            return this._buildURL(modelName) + `/${provider}/moderators/`;
```

Comment:
Minor; would this be better as one template literal instead of `+` and a template literal?
```javascript
return `${this._buildURL(modelName)}/${provider}/moderators/`;
```